snuba/query/validation/validators.py

```diff
@@ -434,3 +438,76 @@ def validate(self, query: Query, alias: Optional[str] = None) -> None:
                                     logger.warning(
                                         f"{lhs} requires datetime conditions: '{param.value}' is not a valid datetime"
                                     )
+
+
+class IllegalAggregateInConditionValidator(QueryValidator):
+    """
+    Ensures that aggregate functions are not used in WHERE clause of query.
+    """
+
+    def __init__(self) -> None:
+        # This is not an exhaustive list of all aggregate functions,
+        # but should be sufficient for catching most invalid queries
+        common_aggregate_functions = [
+            "count",
+            "min",
+            "max",
+            "sum",
+            "avg",
+            "last",
+            "uniq",
+        ]
+        self.aggregate_function_names = [
+            String(func_name) for func_name in common_aggregate_functions
+        ]
+        self.aggregate_function_names.extend(
+            [String(f"{func_name}If") for func_name in common_aggregate_functions]
+        )
+
+    def validate(self, query: Query, alias: Optional[str] = None) -> None:
+        def find_illegal_aggregate_functions(
+            expression: Expression,
+        ) -> list[MatchResult]:
+            matches: list[MatchResult] = []
+            match = FunctionCallMatcher(
+                function_name=Param(
+                    "aggregate",
+                    Or(self.aggregate_function_names),
+                ),
+                with_optionals=True,
+            ).match(expression)
+            if match is not None:
+                matches.append(match)
+
+            match = FunctionCallMatcher(
```

Comment:
As far as I understand, this will match on any boolean function. Why is it necessary?

Reply:
It matches any boolean function in order to traverse the conditions tree down to the aggregation function. Conditions are almost always nested and chained together with boolean conditions.